### PR TITLE
Allow sorting exercises by start date

### LINKED ISSUE #2579: Allow sorting exercises by start date
### Is your feature request related to a problem?
Because points do not actually count something in your subject, we set all due dates far in the future on the same date.
Screenshot from 2020-12-20 16-55-36
The exercises are not at all in order... That is bad students don't know what exercise belongs to which week. Sorting by start/release date would be way more appropriate.

### Describe the solution you'd like
Let the user choose how to sort the exercises

### PR DESCRIPTION
**Pull request by axel1200**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Client: I added multiple integration tests (Jest) related to the features (with a high test coverage)
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context
- The user is only able to sort for dueDates, but it should also be possible to sort for releaseDates

### Description
- Added nur dropdown with a new sortingtype, releaseDate
- added tests for this component
- **Removed code which should be unused, please also double check this in the code review**

### Steps for Testing

1. Log in to Artemis
2. Navigate to Course overview
3. See if exercises are ordered correctly after dueDate
4. Vary with the sorting options (Newest, Oldest, DueDate, ReleaseDate) and make sure exercises are correctly ordered based on the chosen sorting type (see screenshots). Play around with the sorting button and make sure no errors occur and the sorting is always correct.

### Test Coverage
course-exercises.component.ts: 20% -> 97,93%

### Screenshots
Screenshot 2021-01-15 at 14 02 12

### CODE REVIEW

**Comment by axel1200** [comment]
if that should need test maybe you can link me what kind of tests and an example in artemis repo and some kind of tutorial. I am sure I will figure it out

File: src/main/webapp/app/overview/course-exercises/course-exercises.component.ts [hunk 1]
```diff
@@ -190,7 +241,7 @@ export class CourseExercisesComponent implements OnInit, OnChanges, OnDestroy {
                     groupedExercises[dateIndex] = {
                         start: moment(dateValue).startOf('week'),
                         end: moment(dateValue).endOf('week'),
-                        isCollapsed: dateValue.isBefore(moment(), 'week'),
+                        isCollapsed: this.sortingAttribute === this.DUE_DATE ? dateValue.isBefore(moment(), 'week') : dateValue.isBefore(moment().subtract(4, 'weeks'), 'week'),
```
Comment (axel1200) [inline src/main/webapp/app/overview/course-exercises/course-exercises.component.ts:244]: Can be discussed how to handle this

**Review by airelawaleria — APPROVED** [review]
Worked for me on ts1

**Review by JohannesWeiss — APPROVED** [review]
Could not find issues on ts1, works as intended.

**Review by simonzilker — DISMISSED** [review]

**Review by marvinchimeka — APPROVED** [review]

**Review by EneaGore — DISMISSED** [review]

**Review by valeriebucher — APPROVED** [review]

**Review by DenizShabani — APPROVED** [review]

**Review by FlorianStang — DISMISSED** [review]
Tested on TS1, everything worked as intended. Nice Feature

**Review by philgabr — DISMISSED** [review]
Approve

**Review by sebastiantpeter — APPROVED** [review]
Tested on ts1 with the available sorting options. Works as expected.

**Comment by krusche** [comment]
the check client-style still fails. Please check the lint and prettier errors

**Review by tiffca — DISMISSED** [review]
Tested successfully

**Review by JannisBittlmayer — DISMISSED** [review]
Works as intended

**Review by nicolasruscher — DISMISSED** [review]
Looked over the code, the changes look good to me 👍

File: src/test/javascript/spec/component/course/course-exercises.component.spec.ts [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+                spyOn(translateService, 'onLangChange').and.returnValue(of(new Subject()));
+                spyOn(localStorageService, 'retrieve').and.returnValue('OVERDUE,NEEDS_WORK');
+                courseCalculationSpy = stub(courseCalculation, 'getCourse').returns(course);
+
+                fixture.detectChanges();
+            });
+    });
+
+    afterEach(() => {
+        component.ngOnDestroy();
```
Comment (stefanwaldhauser) [inline src/test/javascript/spec/component/course/course-exercises.component.spec.ts:108]: This is not needed in tests.

File: src/test/javascript/spec/component/course/course-exercises.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+        component.ngOnChanges();
+        expect(component.nextRelevantExercise).to.deep.equal(exercise);
+    });
+
+    it('should reorder all exercises', () => {
+        component.sortingOrder = ExerciseSortingOrder.DESC;
+        const sortingButton = fixture.debugElement.query(By.css('#flip'));
+        expect(sortingButton).to.exist;
+        sortingButton.nativeElement.click();
+        expect(component.sortingOrder).to.equal(ExerciseSortingOrder.ASC);
```
Comment (stefanwaldhauser) [inline src/test/javascript/spec/component/course/course-exercises.component.spec.ts:137]: You are not really asserting that the sorting worked here. I would expect to see some test that asserts that the sorting really happened

**Review by stefanwaldhauser — CHANGES_REQUESTED** [review]
some small change requests

File: src/test/javascript/spec/component/course/course-exercises.component.spec.ts (hunk 3 above)
Comment (FuchsDominik) [reply src/test/javascript/spec/component/course/course-exercises.component.spec.ts:156]: Extended this test with some further expectations and calls, hope this hits the point what you expected (80405fbc1dea9ebf80621ffc70f83f0569ecd7bb)

**Review by philippbauch — APPROVED** [review]
The feature works as described and the code also looks good to me!

**Review by stefanwaldhauser — APPROVED** [review]
Code looks good now :) Gj

**Review by airelawaleria — APPROVED** [review]

**Review by simonzilker — APPROVED** [review]

**Review by krusche — APPROVED** [review]

**Pull request merged**